### PR TITLE
Remove unnecessary assignments

### PR DESCRIPTION
**Pull request by Rubonnek**
This patch mostly simplifies some `Vector` or `HashMap` accesses detected in #84445.

### CODE REVIEW

File: scene/main/resource_preloader.cpp [hunk 1]
```diff
@@ -40,10 +40,9 @@ void ResourcePreloader::_set_resources(const Array &p_data) {
 	ERR_FAIL_COND(names.size() != resdata.size());
 
 	for (int i = 0; i < resdata.size(); i++) {
-		String name = names[i];
 		Ref<Resource> resource = resdata[i];
 		ERR_CONTINUE(!resource.is_valid());
-		resources[name] = resource;
+		resources[names[i]] = resource;
 
 		//add_resource(name,resource);
```
Comment (AThousandShips) [inline scene/main/resource_preloader.cpp:47]:
```suggestion
		//add_resource(names[i],resource);
```
To prevent confusion if this comment is restored

**Review by AThousandShips — APPROVED** [review]
These looks good (just confirming my own suggestions so someone else should confirm as well)

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 2]
```diff
@@ -4820,16 +4820,14 @@ void VisualShaderEditor::_varying_create() {
 	add_varying_dialog->hide();
 }
 
-void VisualShaderEditor::_varying_name_changed(const String &p_text) {
-	String name = p_text;
-
-	if (!name.is_valid_identifier()) {
+void VisualShaderEditor::_varying_name_changed(const String &p_name) {
```
Comment (YuriSizov) [inline editor/plugins/visual_shader_editor_plugin.cpp:4823]: If you rename a parameter here, you also need to rename it in the header.

File: scene/resources/visual_shader.cpp [hunk 3]
```diff
@@ -4217,11 +4217,10 @@ void VisualShaderNodeGroupBase::set_inputs(const String &p_inputs) {
 
 		int port_idx = arr[0].to_int();
 		int port_type = arr[1].to_int();
-		String port_name = arr[2];
 
 		Port port;
 		port.type = (PortType)port_type;
-		port.name = port_name;
+		port.name = arr[2];
 		input_ports[port_idx] = port;
```
Comment (YuriSizov) [inline scene/resources/visual_shader.cpp:4224]: These two do not look like an improvement to me. Sure, we don't need to cast anything here for the string, but the way the code is laid out makes it very easy to see the composition of the parsed input string before it is assigned to the corresponding fields and properties. I would keep it as is.

**Comment by YuriSizov** [comment]
Looks fine to me, aside from the two things that I've mentioned. Once addressed, this can be merged.

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**